pkg/controller/seed-controller-manager/etcdrestore/etcd_restore_controller.go

```diff
@@ -331,7 +335,32 @@ func (r *Reconciler) updateCluster(ctx context.Context, cluster *kubermaticv1.Cl
 	if reflect.DeepEqual(oldCluster, cluster) {
 		return nil
 	}
-	return r.Client.Patch(ctx, cluster, ctrlruntimeclient.MergeFrom(oldCluster))
+
+	if !reflect.DeepEqual(oldCluster.Status, cluster.Status) {
+		return errors.New("updateCluster must not change cluster status")
+	}
+
+	if err := r.Patch(ctx, cluster, ctrlruntimeclient.MergeFrom(oldCluster)); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (r *Reconciler) updateClusterStatus(ctx context.Context, cluster *kubermaticv1.Cluster, modify func(*kubermaticv1.Cluster)) error {
+	oldCluster := cluster.DeepCopy()
+	modify(cluster)
+	if reflect.DeepEqual(oldCluster, cluster) {
+		return nil
+	}
+
+	if !reflect.DeepEqual(oldCluster.Status, cluster.Status) {
+		if err := r.Status().Patch(ctx, cluster, ctrlruntimeclient.MergeFrom(oldCluster)); err != nil {
+			return err
+		}
+	}
+
+	return nil
```

Comment:
Looks like other controllers have exactly the same code to update the kubermaticv1.Cluster, maybe let's move it to some helpers

Reply:
No further refactoring please. This is waaaaaay to large of a PR already.